fmriprep/cli/run.py

```diff
@@ -212,78 +226,78 @@ def create_workflow(opts):
-                  'may need to adjust your "File sharing" preferences.'.format(bids_dir))
-            sys.exit(1)
-    else:
-        subject_list = [sub[4:] if sub.startswith('sub-') else sub for sub in subject_list]
+    # Set up directories
+    output_dir = op.abspath(opts.output_dir)
+    log_dir = op.join(opts.output_dir, 'fmriprep', 'logs')
+    work_dir = op.abspath(opts.work_dir)
 
-    logger.info('Subject list: %s', ', '.join(subject_list))
+    # Check and create output and working directories
+    os.makedirs(output_dir, exist_ok=True)
+    os.makedirs(log_dir, exist_ok=True)
+    os.makedirs(work_dir, exist_ok=True)
 
-    # Build main workflow
-    reportlets_dir = op.join(op.abspath(opts.work_dir), 'reportlets')
-    bids_dir = op.abspath(opts.bids_dir)
-    if not opts.reports_only or opts.write_graph:
-        fmriprep_wf = init_fmriprep_wf(subject_list=subject_list,
-                                       task_id=opts.task_id,
-                                       run_uuid=run_uuid,
-                                       ignore=opts.ignore,
-                                       debug=opts.debug,
-                                       anat_only=opts.anat_only,
-                                       longitudinal=opts.longitudinal,
-                                       omp_nthreads=omp_nthreads,
-                                       skull_strip_ants=opts.skull_strip_ants,
-                                       reportlets_dir=reportlets_dir,
-                                       output_dir=output_dir,
-                                       bids_dir=bids_dir,
-                                       freesurfer=opts.freesurfer,
-                                       output_spaces=opts.output_space,
-                                       template=opts.template,
-                                       output_grid_ref=opts.output_grid_reference,
-                                       hires=opts.hires,
-                                       bold2t1w_dof=opts.bold2t1w_dof,
-                                       fmap_bspline=opts.fmap_bspline,
-                                       fmap_demean=opts.fmap_no_demean,
-                                       use_syn=opts.use_syn_sdc,
-                                       force_syn=opts.force_syn,
-                                       use_aroma=opts.use_aroma,
-                                       ignore_aroma_err=opts.ignore_aroma_denoising_errors)
-
-        fmriprep_wf.base_dir = op.abspath(opts.work_dir)
+    # Nipype config (logs and execution)
+    ncfg.update_config({
+        'logging': {'log_directory': log_dir, 'log_to_file': True},
+        'execution': {'crashdump_dir': log_dir, 'crashfile_format': 'txt'},
+    })
 
+    # Called with reports only
     if opts.reports_only:
-        if opts.write_graph:
-            fmriprep_wf.write_graph(graph2use="colored", format='svg', simple_form=True)
```

Comment:
One thing I meant is: `--reports-only` means you want just the reports, and it should be independent of `--write-graph`. If you really just want the graph, it wasn't clear that you also needed to say `--reports-only` in the former version. It was like a hidden feature.